chimney-java-collections/src/main/scala/io/scalaland/chimney/javacollections/internal/TransformOrUpcast.scala

```diff
@@ -0,0 +1,22 @@
+package io.scalaland.chimney.javacollections.internal
+
+import io.scalaland.chimney.Transformer
+
+/** Since [[io.scalaland.chimney.Transformer]] is NOT automatically provided for identity/subtype transformation,
+  * and we want to allow such things without enabling whole recursive auto-derivation we use this trick.
+  */
+trait TransformOrUpcast[From, To] {
```

Comment:
Question: shouldn't such structure live in core? Would we reuse it somewhere else? If we want to enure binary compatibility we should decide where to put this utility before releasing it since shoving it around would break binary compatibility.